src/changelog.ts

```diff
@@ -0,0 +1,168 @@
+import * as vscode from 'vscode';
+
+import { NEURO } from '@/constants';
+import { getFence, logOutput } from '@/utils';
+import { ActionData, RCEAction, stripToActions } from '@/neuro_client_helper';
+import { PermissionLevel } from '@/config';
+
+const MEMENTO_KEY = 'lastDeliveredChangelogVersion';
+
+interface ChangelogSection {
+    version: string;
+    body: string;
+}
+
+export const changelogActions = {
+    read_changelog: {
+        name: 'read_changelog',
+        description: 'Get changelog entries starting from a specified version. If fromVersion is omitted, any new entries after the last read_changelog command are read.',
+        schema: {
+            type: 'object',
+            properties: {
+                fromVersion: { type: 'string', description: 'Version (e.g., 2.2.1) to start including entries from, inclusive.' },
+            },
+            additionalProperties: false,
+        },
+        permissions: [],
+        defaultPermission: PermissionLevel.COPILOT,
+        handler: handleReadChangelog,
+        promptGenerator: (actionData: ActionData) => actionData.params?.fromVersion
+            ? `Send all changelog entries starting from version ${actionData.params.fromVersion} (inclusive).`
```

Comment:
```suggestion
            ? `read all changelog entries starting from version ${actionData.params.fromVersion} (inclusive).`
```

`promptGenerator` must follow the scheme of "Neuro wants to (x)", where "(x)" is replaced with the `promptGenerator` string.